### PR TITLE
Add support for Fedora 31, 32, and 33 and remove support versions older than 29

### PR DESCRIPTION
**Pull request by jmciver**
This pull request is comprised of three commits. 

Resolves Ruby gem dependency issue: 
The redcarpet gem is now used during linting and generation of REFERENCE.md.

Add support for Fedora version 31, 32, and 33:
Additionally, the commit adds support for warning about unsupported versions of Fedora. This was done to suppress warning messages concerning Fedora not being supported. I adjusted the conditional check to warn on non-explicitly supported versions of operating systems in the RedHat family.

Remove support for EOL Fedora versions older than 30:
This was done in an attempt to tidy up. Fedora versions are 2 a year and things can get messy if old versions hang around. However, reality is that old versions linger so two EOL versions, 30 and 31, are still being supported.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -92,11 +92,20 @@
...
+    if ($::facts['os']['name'] == 'Amazon') {
+      if !($::facts['os']['release']['major'] in ['2']) {
+        warning("osname Amazon's os.release.major is <${::facts['os']['release']['major']}> and must be 2.")
+      }
+    } elsif ($::facts['os']['name'] in ['RedHat', 'CentOS']) {
+      if !($::facts['os']['release']['major'] in ['6', '7', '8']) {
+        warning("osname RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 6, 7 or 8.")
+      }
+    } elsif ($::facts['os']['name'] == 'Fedora') {
+      if !($::facts['os']['release']['major'] in ['29', '30', '31', '32', '33']) {
```
Comment (ghoneycutt) [inline manifests/init.pp:104]: `29` should not be here since its data was removed.

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

**Comment by ghoneycutt** [comment]
Awesome PR @jmciver !

Think the `29` was a typo, once that's cleared up I'll merge.

File: manifests/init.pp (hunk 1 above)
Comment (jmciver) [reply manifests/init.pp]: Good catch! I have removed the unnecessary `29` from the array declaration, squashed, and pushed.

**Comment by jmciver** [comment]
Thanks catching that @ghoneycutt ! I have fixed and pushed.

**Comment by jmciver** [comment]
Sorry about re-request the review; I was playing with the interface and there seems to be no way to deactivate it...

**Review by ghoneycutt — APPROVED** [review]

**Comment by dploeger** [comment]
@ghoneycutt Thanks for merging. Any ideas when this will be released on the forge?

**Pull request merged**